src/response.ts

```diff
@@ -232,22 +233,42 @@ document.addEventListener('DOMContentLoaded', async () => {
               _ =>
                 // 1.0 final status
                 showFinalStatusResult(transactionStatus.data.idStatus),
-              _ =>
-                transactionStatus.data.methodUrl !== '' &&
-                (transactionStatus.data.xpayHtml === '' || transactionStatus.data.xpayHtml === undefined)
-                  ? // 1.2 METHOD step 3ds2
+              _ => {
+                switch (nextTransactionStep(transactionStatus)) {
+                  // 1.1 METHOD step 3ds2
+                  case 'method': {
                     fromNullable(transactionStatus.data.threeDSMethodData).fold(none, threeDSMethodData => {
                       sessionStorage.setItem('threeDSMethodData', threeDSMethodData);
                       return start3DS2MethodStep(
                         transactionStatus.data.methodUrl,
                         transactionStatus.data.threeDSMethodData,
                         createIFrame(document.body, 'myIdFrame', 'myFrameName'),
                       );
-                    })
-                  : // 1.1 XPAY step 3ds2
+                    });
+                    break;
+                  }
+                  // 1.2 Challenge step 3ds2
+                  case 'challenge': {
```

Comment:
Maybe we could add a comment to explain that this challenge step is never executed after a successfull method step, since in that case the challenge is started inside the message handler. As a result, this challenge case manages flows that don't start with a method